src/overmap.cpp

```diff
@@ -162,12 +162,21 @@ bool is_river(const oter_id &ter)
 bool is_ot_type(const std::string &otype, const oter_id &oter)
 {
     const size_t compare_size = otype.size();
-    if (compare_size > oter.size()) {
-        return false;
-    } else {
-        return std::string(oter).compare(0, compare_size, otype ) == 0;
-    }
 
+    if (compare_size <= oter.size()) {
+        const auto oter_str = std::string(oter);
+        if (oter_str.compare(0, compare_size, otype) == 0) {
+            // check if it's a full match
+            if (compare_size == oter.size()) {
+                return true;
+            }
+            // only ok for partial if next char is an underscore
+            if (oter_str.at(compare_size) == '_') {
```

Comment:
Shouldn't `compare_size <= oter.size()` and `compare_size == oter.size()` ensure `oter_str.at(compare_size)` is never `out_of_range`?